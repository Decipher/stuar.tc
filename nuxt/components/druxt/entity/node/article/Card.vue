<template>
  <DuiCard
    v-if="entity.attributes.path"
    :title="entity.attributes.title"
    :to="entity.attributes.path.alias"
  >
    <div class="flex-grow flex gap-1">
      <!-- Date badge -->
      <DuiBadge class="mb-3" size="sm">
        {{ $moment(entity.attributes.created).format('YYYY.MM.DD') }}
      </DuiBadge>

      <template v-if="entity.included">
        <!-- Type badge -->
        <DuiBadge class="mb-5" size="sm" type="primary">{{
          entity.included.find((o) => o.type === 'taxonomy_term--article_type')
            .attributes.name
        }}</DuiBadge>

        <!-- Category badge -->
        <DuiBadge class="mb-5" size="sm" type="secondary">{{
          entity.included.find(
            (o) => o.type === 'taxonomy_term--article_category'
          ).attributes.name
        }}</DuiBadge>
      </template>
    </div>

    <!-- eslint-disable-next-line vue/no-v-html -->
    <div v-if="!mini" class="prose" v-html="description" />
  </DuiCard>

  <div v-else />
</template>

<script>
import { DruxtEntityMixin } from 'druxt-entity'
import ellipsize from 'ellipsize'

export default {
  mixins: [DruxtEntityMixin],

  props: {
    mini: {
      type: Boolean,
      default: false,
    },
  },

  computed: {
    description: ({ $md, fields }) =>
      ellipsize(
        $md.render(fields.field_description.data),
        fields.field_description.schema.settings.display.trim_length
      ),
  },

  druxt: {
    query: {
      include: ['field_article_category', 'field_article_type'],
      fields: [
        [
          'created',
          'field_article_category',
          'field_description',
          'path',
          'title',
        ],
        ['taxonomy_term--article_category', ['name']],
        ['taxonomy_term--article_type', ['name']],
      ],
    },
  },
}
</script>
