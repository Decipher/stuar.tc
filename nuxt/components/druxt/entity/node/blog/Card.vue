<template>
  <DuiCard
    v-if="entity.attributes.path"
    :title="entity.attributes.title"
    :to="entity.attributes.path.alias"
  >
    <div class="flex gap-1">
      <!-- Date badge -->
      <DuiBadge class="mb-3" size="sm">
        {{ $moment(entity.attributes.created).format('YY.MM.DD') }}
      </DuiBadge>

      <!-- Category badge -->
      <DuiBadge class="mb-5" size="sm" type="primary">{{ entity.included.find((o) => o.type === 'taxonomy_term--blog').attributes.name }}</DuiBadge>
    </div>
    <div class="prose" v-html="entity.attributes.field_description.processed" />
  </DuiCard>
</template>

<script>
import { DruxtEntityMixin } from 'druxt-entity'
export default {
  mixins: [DruxtEntityMixin],
  druxt: {
    query: {
      include: ['field_blog_category'],
      fields: [
        ['created', 'field_blog_category', 'field_description', 'path', 'title'],
        ['taxonomy_term--blog', ['name']]
      ]
    }
  }
}
</script>
