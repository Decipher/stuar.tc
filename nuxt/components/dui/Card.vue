<template>
  <div class="card h-full flex lg:card-side bordered shadow hover:shadow-lg">
    <div class="card-body">
      <h2 class="card-title" v-text="title" />
      <slot />
      <div class="card-actions">
        <NuxtLink :to="to" class="btn btn-primary">Open post</NuxtLink>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  props: {
    mini: {
      type: Boolean,
      default: false
    },
    title: {
      type: String,
      required: true,
    },
    to: {
      type: String,
      required: true,
    }
  }
}
</script>
