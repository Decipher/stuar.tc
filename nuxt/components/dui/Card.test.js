import { mount } from '@vue/test-utils'
import Card from './Card.vue'

const stubs = ['NuxtLink']

describe('Card', () => {
  test('is a Vue instance', () => {
    const props = {
      title: 'test',
      to: '/',
    }
    const wrapper = mount(Card, { props, stubs })
    expect(wrapper.vm).toBeTruthy()

    // Expect HTML to match snapshot.
    expect(wrapper.html()).toMatchSnapshot()
  })
})
