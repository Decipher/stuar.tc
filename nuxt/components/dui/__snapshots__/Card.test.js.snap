// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card is a Vue instance 1`] = `
"<div class=\\"card h-full flex lg:card-side bordered shadow hover:shadow-lg\\">
  <div class=\\"card-body\\">
    <h2 class=\\"card-title\\"></h2>
    <div class=\\"card-actions\\">
      <nuxtlink-stub class=\\"btn btn-primary\\">Open post</nuxtlink-stub>
    </div>
  </div>
</div>"
`;
